### ISSUE TITLE
ios setup failing for me

### ISSUE THREAD

**Issue by joeblew99** [therecipe/qt#219]
Anything stand out ?

in gallery example it created an "ios_minimal" folder but nothing else.

````

x-MacBook-Pro:qt apple$ $GOPATH/bin/qtsetup ios && $GOPATH/bin/qtsetup ios-simulator
INFO[0000] running setup/prep                           
INFO[0000] running setup/check ios                      
INFO[0000] VERSION: 0.0.4                               
INFO[0000] GOOS: darwin                                 
INFO[0000] GOARCH: amd64                                
INFO[0000] GOVERSION: go1.7.4                           
INFO[0000] GOROOT: /usr/local/go                        
INFO[0000] GOPATH: /Users/apple/workspace/go            
INFO[0000] GOBIN: /Users/apple/workspace/go/bin         
INFO[0000] HASH: d874b0a4b22e34a1cc253218e2f4ca09c9fe686d 
INFO[0000] QT_VERSION: 5.7.0                            
INFO[0000] QT_VERSION_MAJOR: 5.7                        
INFO[0000] QT_DIR: /Users/apple/Qt5.7.0                 
INFO[0000] QT_STUB: false                               
INFO[0000] QT_HOMEBREW:                                 
INFO[0000] IsHomeBrewQtDir: false                       
INFO[0000] UseHomebrew: false                           
INFO[0000] QT_DARWIN_DIR: /Users/apple/Qt5.7.0/5.7/clang_64 
INFO[0000] XCODE_DIR: /Applications/Xcode.app           
INFO[0000] MACOS_SDK_DIR: MacOSX10.12.sdk               
INFO[0000] IPHONEOS_SDK_DIR: iPhoneOS10.2.sdk           
INFO[0000] IPHONESIMULATOR_SDK_DIR: iPhoneSimulator10.2.sdk 
INFO[0000] running setup/generate                       
INFO[0000] running setup/install ios                    
INFO[0000] running setup/test ios (~5min)               
INFO[0000] testing qml/application                      
INFO[0043] testing qml/drawer_nav_x                     
INFO[0079] testing qml/gallery                          
ERRO[0089] failed to run command                         cmd="go build -p 4 -v -ldflags=\"-w\" -o /Users/apple/workspace/go/src/github.com/therecipe/qt/internal/examples/qml/gallery/deploy/ios_minimal/libgo.a -tags=\"minimal\" \"ios\" -i -installsuffix ios -pkgdir /Users/apple/workspace/go/pkg/darwin_arm64_ios -buildmode c-archive" error="exit status 2" func=RunCmd name="build for ios on darwin"
runtime/internal/sys
runtime/internal/atomic
runtime
errors
internal/race
sync/atomic
math
unicode/utf8
github.com/therecipe/qt/internal/android/cgo
unicode
sync
runtime/cgo
io
syscall
strconv
strings
reflect
time
os
github.com/therecipe/qt/quickcontrols2
fmt
# github.com/therecipe/qt/quickcontrols2
/tmp/go-build171707462/github.com/therecipe/qt/quickcontrols2/_obj/_cgo_main.c:2:23: warning: unused parameter 'fn' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/quickcontrols2/_obj/_cgo_main.c:2:61: warning: unused parameter 'a' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/quickcontrols2/_obj/_cgo_main.c:2:68: warning: unused parameter 'c' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/quickcontrols2/_obj/_cgo_main.c:2:85: warning: unused parameter 'ctxt' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/quickcontrols2/_obj/_cgo_main.c:4:41: warning: unused parameter 'ctxt' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/quickcontrols2/_obj/_cgo_main.c:6:26: warning: unused parameter 'a' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/quickcontrols2/_obj/_cgo_main.c:6:33: warning: unused parameter 'c' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/quickcontrols2/_obj/_cgo_main.c:7:23: warning: unused parameter 'a' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/quickcontrols2/_obj/_cgo_main.c:7:30: warning: unused parameter 'c' [-Wunused-parameter]
log
github.com/therecipe/qt
github.com/therecipe/qt/core
# github.com/therecipe/qt/core
/tmp/go-build171707462/github.com/therecipe/qt/core/_obj/_cgo_main.c:2:23: warning: unused parameter 'fn' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/core/_obj/_cgo_main.c:2:61: warning: unused parameter 'a' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/core/_obj/_cgo_main.c:2:68: warning: unused parameter 'c' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/core/_obj/_cgo_main.c:2:85: warning: unused parameter 'ctxt' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/core/_obj/_cgo_main.c:4:41: warning: unused parameter 'ctxt' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/core/_obj/_cgo_main.c:6:26: warning: unused parameter 'a' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/core/_obj/_cgo_main.c:6:33: warning: unused parameter 'c' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/core/_obj/_cgo_main.c:7:23: warning: unused parameter 'a' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/core/_obj/_cgo_main.c:7:30: warning: unused parameter 'c' [-Wunused-parameter]
# github.com/therecipe/qt/quickcontrols2
Undefined symbols for architecture arm64:
  "QQmlEngine::QQmlEngine(QObject*)", referenced from:
      QQuickStyleSpec::resolve() in libQt5QuickControls2.a(qquickstyle.o)
  "QQmlEngine::~QQmlEngine()", referenced from:
      QQuickStyleSpec::resolve() in libQt5QuickControls2.a(qquickstyle.o)
  "QQuickItem::windowChanged(QQuickWindow*)", referenced from:
      QQuickStyleAttached::QQuickStyleAttached(QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
      QQuickStyleAttached::~QQuickStyleAttached() in libQt5QuickControls2.a(qquickstyleattached.o)
  "QQuickItem::staticMetaObject", referenced from:
      QQuickStyleAttached::QQuickStyleAttached(QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
      QQuickStyleAttached::itemWindowChanged(QQuickWindow*) in libQt5QuickControls2.a(qquickstyleattached.o)
      QQuickStyleAttached::~QQuickStyleAttached() in libQt5QuickControls2.a(qquickstyleattached.o)
      findParentStyle(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
      findChildStyles(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
  "QQuickPopup::staticMetaObject", referenced from:
      QQuickStyleAttached::QQuickStyleAttached(QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
      findParentStyle(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
  "QQmlMetaType::isModule(QString const&, int, int)", referenced from:
      QQuickStyle::setStyle(QString const&) in libQt5QuickControls2.a(qquickstyle.o)
  "QQuickWindow::staticMetaObject", referenced from:
      findParentStyle(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
      findChildStyles(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
  "QQuickItemPrivate::addItemChangeListener(QQuickItemChangeListener*, QFlags<QQuickItemPrivate::ChangeType>)", referenced from:
      QQuickStyleAttached::QQuickStyleAttached(QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
  "QQuickItemPrivate::removeItemChangeListener(QQuickItemChangeListener*, QFlags<QQuickItemPrivate::ChangeType>)", referenced from:
      QQuickStyleAttached::~QQuickStyleAttached() in libQt5QuickControls2.a(qquickstyleattached.o)
  "QtQml::qmlAttachedPropertiesObject(int*, QObject const*, QMetaObject const*, bool)", referenced from:
      findParentStyle(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
      findChildStyles(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
  "QtQml::qmlEngine(QObject const*)", referenced from:
      findParentStyle(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
  "QQmlFile::isLocalFile(QString const&)", referenced from:
      QQuickStyleSpec::path() in libQt5QuickControls2.a(qquickstyle.o)
  "QQmlFile::urlToLocalFileOrQrc(QString const&)", referenced from:
      QQuickStyleSpec::path() in libQt5QuickControls2.a(qquickstyle.o)
  "QQmlEngine::importPathList() const", referenced from:
      QQuickStyleSpec::resolve() in libQt5QuickControls2.a(qquickstyle.o)
  "QQuickItem::childItems() const", referenced from:
      findChildStyles(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
  "QQuickItem::parentItem() const", referenced from:
      findParentStyle(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
  "QQuickItem::window() const", referenced from:
      findParentStyle(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
  "QQuickPopup::popupItem() const", referenced from:
      QQuickStyleAttached::QQuickStyleAttached(QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
      findParentStyle(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
  "QQuickWindow::contentItem() const", referenced from:
      findChildStyles(QMetaObject const*, QObject*) in libQt5QuickControls2.a(qquickstyleattached.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
# github.com/therecipe/qt/core
../../../../core/core-minimal.cpp:170:40: warning: unused parameter 'ptr' [-Wunused-parameter]
../../../../core/core-minimal.cpp:186:46: warning: unused parameter 'ptr' [-Wunused-parameter]
../../../../core/core-minimal.cpp:343:42: warning: unused parameter 'ptr' [-Wunused-parameter]
../../../../core/core-minimal.cpp:977:47: warning: unused parameter 'ptr' [-Wunused-parameter]
../../../../core/core-minimal.cpp:997:47: warning: unused parameter 'ptr' [-Wunused-parameter]
../../../../core/core-minimal.cpp:1012:47: warning: unused parameter 'ptr' [-Wunused-parameter]
../../../../core/core-minimal.cpp:1032:53: warning: unused parameter 'ptr' [-Wunused-parameter]
../../../../core/core-minimal.cpp:1047:53: warning: unused parameter 'ptr' [-Wunused-parameter]
github.com/therecipe/qt/gui
github.com/therecipe/qt/qml
# github.com/therecipe/qt/qml
/tmp/go-build171707462/github.com/therecipe/qt/qml/_obj/_cgo_main.c:2:23: warning: unused parameter 'fn' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/qml/_obj/_cgo_main.c:2:61: warning: unused parameter 'a' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/qml/_obj/_cgo_main.c:2:68: warning: unused parameter 'c' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/qml/_obj/_cgo_main.c:2:85: warning: unused parameter 'ctxt' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/qml/_obj/_cgo_main.c:4:41: warning: unused parameter 'ctxt' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/qml/_obj/_cgo_main.c:6:26: warning: unused parameter 'a' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/qml/_obj/_cgo_main.c:6:33: warning: unused parameter 'c' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/qml/_obj/_cgo_main.c:7:23: warning: unused parameter 'a' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/qml/_obj/_cgo_main.c:7:30: warning: unused parameter 'c' [-Wunused-parameter]
# github.com/therecipe/qt/gui
/tmp/go-build171707462/github.com/therecipe/qt/gui/_obj/_cgo_main.c:2:23: warning: unused parameter 'fn' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/gui/_obj/_cgo_main.c:2:61: warning: unused parameter 'a' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/gui/_obj/_cgo_main.c:2:68: warning: unused parameter 'c' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/gui/_obj/_cgo_main.c:2:85: warning: unused parameter 'ctxt' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/gui/_obj/_cgo_main.c:4:41: warning: unused parameter 'ctxt' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/gui/_obj/_cgo_main.c:6:26: warning: unused parameter 'a' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/gui/_obj/_cgo_main.c:6:33: warning: unused parameter 'c' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/gui/_obj/_cgo_main.c:7:23: warning: unused parameter 'a' [-Wunused-parameter]
/tmp/go-build171707462/github.com/therecipe/qt/gui/_obj/_cgo_main.c:7:30: warning: unused parameter 'c' [-Wunused-parameter]
# github.com/therecipe/qt/qml
../../../../qml/qml-minimal.cpp:36:46: warning: unused parameter 'ptr' [-Wunused-parameter]
../../../../qml/qml-minimal.cpp:143:57: warning: unused parameter 'ptr' [-Wunused-parameter]
../../../../qml/qml-minimal.cpp:159:48: warning: unused parameter 'ptr' [-Wunused-parameter]
../../../../qml/qml-minimal.cpp:185:58: warning: unused parameter 'ptr' [-Wunused-parameter]
# github.com/therecipe/qt/gui
../../../../gui/gui-minimal.cpp:47:60: warning: unused parameter 'ptr' [-Wunused-parameter]
../../../../gui/gui-minimal.cpp:166:58: warning: unused parameter 'ptr' [-Wunused-parameter]
../../../../gui/gui-minimal.cpp:217:52: warning: unused parameter 'ptr' [-Wunused-parameter]

````

**Comment by therecipe** [comment]
Hey

I will look into this.
It's probably only a minor linking problem.

**Comment by joeblew99** [comment]
Thanks - would be supper cool.

I am at the stage of trying to get the gallery example running on iOS to just prove the architecture.

**Comment by therecipe** [comment]
Should work now with: https://github.com/therecipe/qt/commit/b1cb04c1a009ae7b807f7ed7e1ddfa86d3f8606a

drawer_nav_x doesn't seem to work though, but I will fix that with the update to Qt 5.8.0.

**Comment by joeblew99** [comment]
i am good using IOS and ios-simulate now.

SO will close this....
